### ISSUE TITLE
Remove Traditional/Basic Workflow from codebase and database

### ISSUE THREAD

**Issue by tdonohue** [DSpace/DSpace#2995]
DSpace 7 will only support Configurable Workflow.  This was decided in [DS-3851](https://jira.lyrasis.org/browse/DS-3851) and implemented as part of https://github.com/DSpace/DSpace/pull/2312

However, the older "traditional" or "basic" workflow still exists in the codebase (and in the form of database tables).  These will need to be removed.

Changes required include:
* Removal of all [`org.dspace.workflowbasic.*` classes](https://github.com/DSpace/DSpace/tree/main/dspace-api/src/main/java/org/dspace/workflowbasic) and [tests](https://github.com/DSpace/DSpace/tree/main/dspace-api/src/test/java/org/dspace/workflowbasic)
   * This may require touching/refactoring other code as well
   * Any prior `if` code checks which check if XmlWorkflowService is enabled also should now be removable. There's only one WorkflowService enabled in DSpace.
* Removal of all database tables/columns related to Basic/Traditional Workflow.  This includes:
   * Dropping/removing all `workflow_step_x` columns from the `collection` table
   * Dropping/deleting the `tasklistitem`, `workflow_item` tables
   * (We also should check to see if any groups or resource policies are specific to Basic/Traditional Workflow and therefore need cleanup.  However, I believe these should be reused by the new Configurable Workflow?)
   * Some early (but incomplete) work on this database cleanup can be found at https://github.com/DSpace/DSpace/pull/2268

See also https://jira.lyrasis.org/browse/DS-3041

**Comment by abollini** [comment]
just a comment to share some initial investigation that I made on this task.
It is not so easy as it could look like. The first problem is that it is possible that some institution has adopted the configurable workflow before, this *could* make a bit more complicated the database cleanup (flyway)
The second problem is about the test classes
org.dspace.workflowbasic.BasicWorkflowAuthorizationIT
org.dspace.workflowbasic.BasicWorkflowAuthorizationRolesIT
they bring a lot of useful test around security of the workflow that are not available for the configurable workflow (as far as I can see) and are not immediately "upgradeable" to rely with it.

Depending on what we want to do with the test (remove and forget or reimplement) I guess that this task could requires from 1-2 days to up 5-7 days to be implemented

**Comment by tdonohue** [comment]
@abollini : Regarding Flyway database "cleanup", keep in mind that we already have a script which automatically checks if a site has previously upgraded to Configurable Workflow before and if *not* upgrades them as part of the 7.0 upgrade.   That script can be found here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V7_0_2018_04_03__Upgrade_Workflow_Policy.java

I don't see any reason to keep around ITs for a Workflow System that is obsolete (even if those tests provide useful checks of that old Workflow System).  Instead, I'd recommend we create a ticket to *recreate* those old tests (as needed) on Configurable Workflow.  However, that task should be estimated separate from this one.

Does that clarify the goals of this ticket?  This ticket should be about removing the old Workflow system *entirely* and, where necessary, logging tickets for the new Workflow system.   So, I think this may align it more with your 1-2 day estimate...is that accurate?

**Comment by abollini** [comment]
let's keep an estimation of **1-2 days** as this would require to verify that an existing dspace 6.3 without xml workflow enabled can migrate cleanly and automatically to version 7 and all the unnecessary basic workflow table are removed. 

The script that you point out already assume that the xml workflow was enabled and do nothing if it wasn't. It could be actually as easy as copy this script and make sure that the migration occur but I'm worried that other flyway migration add fix to the xml workflow table that could be "loss" if the migration run out-of-order

**Comment by tdonohue** [comment]
@abollini : Ok, that sounds fine.  I realized I had linked to the wrong migration in my previous comment.  Here's our migration that can already upgrade users from Basic Workflow to XML Workflow: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/storage/rdbms/xmlworkflow/V6_0_2015_09_01__DS_2701_Enable_XMLWorkflow_Migration.java

Currently, that migration is **already required** in 7.0 since XmlWorkflowServiceImpl is now enabled by default.  So, I don't believe we need to do anything further to ensure everyone runs that migration (other than documenting that users may need to run `ignored` / out-of-order migrations during the upgrade from 6.x to 7.0).  So, there may be nothing left to do here.  But, you are correct, we should test that the process works as expected.

(That said, obviously, we do need a new migration to remove all the old basic workflow tables/columns.)

In any case, I'll add an estimate of **12hrs** (1.5 days) here, since that's the middle of your range.   Thanks!
